### ISSUE TITLE
Test broken on CentOS7.6

### ISSUE THREAD

**Issue by Heidistein** [rubygems/rubygems#2698]
Merge #2367 causes the test to fail on RHEL 7.6.
Reverting the patch fixes it again.

```
  1) Error:
TestGemExtCmakeBuilder#test_self_build:
Gem::InstallError: cmake failed, exit code 1
    /usr/src/redhat/BUILD/ruby-2.6.2/lib/rubygems/ext/builder.rb:99:in `run'
    /usr/src/redhat/BUILD/ruby-2.6.2/lib/rubygems/ext/cmake_builder.rb:10:in `build'
    /usr/src/redhat/BUILD/ruby-2.6.2/test/rubygems/test_gem_ext_cmake_builder.rb:38:in `block in test_self_build'
    /usr/src/redhat/BUILD/ruby-2.6.2/test/rubygems/test_gem_ext_cmake_builder.rb:37:in `chdir'
    /usr/src/redhat/BUILD/ruby-2.6.2/test/rubygems/test_gem_ext_cmake_builder.rb:37:in `test_self_build'

Finished tests in 699.805724s, 28.9466 tests/s, 3263.4972 assertions/s.
20257 tests, 2283814 assertions, 0 failures, 1 errors, 62 skips

ruby -v: ruby 2.6.2p47 (2019-03-13 revision 67232) [x86_64-linux]
make: *** [yes-test-almost] Error 1
error: Bad exit status from /var/tmp/rpm-tmp.2d1xgZ (%check)

```

**Comment by duckinator** [comment]
I'm having trouble reproducing this. I tried CentOS 7.6 in Docker, as well as CentOS 7.6 in a VM.

@Heidistein can you provide the output of running `gem env`, and tell me about any things you suspect may be both relevant and different between my testcase and your system?

## Testcase

Explicitly-installed packages:

1. redhat-lsb-core (only used to provide the output of `lsb_release -d`)
2. git (to clone the repo)
3. gcc (required for certain tests)
4. cmake (because cmake-related tests won't run without it)

Testcase: https://da.gd/2qdHC

Ruby version with this script: `ruby 2.6.2p47 (2019-03-13 revision 67232) [x86_64-linux]`

## Running in Docker

To run the testcase in a Docker container, I ran this command:

```
docker run --rm -it centos:7 bash -c 'curl -L https://da.gd/2qdHC | bash'
```

This was the result:

![Screenshot_20190328_200732](https://user-images.githubusercontent.com/39698/55200580-2852c200-5195-11e9-9851-55372674d340.png)

## Running in a VM

Since I couldn't reproduce in Docker, I decided to try in a virtual machine.

To run the same testcase in a VM, I did a basic CentOS 7 install, and ran:

```
curl -L https://da.gd/2qdHC | bash
```

This was the result:

![Screenshot_20190328_202407](https://user-images.githubusercontent.com/39698/55201032-78328880-5197-11e9-8213-c86586af368a.png)

**Comment by hsbt** [comment]
I couldn't reproduce it with ruby core repository.

https://rubyci.org/logs/rubyci.s3.amazonaws.com/centos7/ruby-trunk/log/20190328T190003Z.log.html.gz

**Comment by simi** [comment]
I can reproduce with docker (like @duckinator) and this script:

```
#!/bin/bash

yum install -y redhat-lsb-core git gcc cmake || exit $?
lsb_release -d || exit $?
curl -LO https://github.com/feedforce/ruby-rpm/releases/download/2.6.2/ruby-2.6.2-1.el7.centos.x86_64.rpm || exit $?
yum install -y ./ruby-2.6.2-1.el7.centos.x86_64.rpm || exit $?
curl -LO https://cache.ruby-lang.org/pub/ruby/2.6/ruby-2.6.2.tar.gz || exit $?
tar -xvzf ruby-2.6.2.tar.gz || exit $?
cd ruby-2.6.2 || exit $?
ruby -Itest test/rubygems/test_gem_ext_cmake_builder.rb  || exit $?
```

It doesn't clone rubygems master, but unpack official 2.6.2 source codes.

**Comment by simi** [comment]
the diff for 2.6.2 and rubygems master is:

```diff
# diff rubygems/test/rubygems/test_gem_ext_cmake_builder.rb ruby-2.6.2/test/rubygems/test_gem_ext_cmake_builder.rb 
28c28
< project(self_build NONE)
---
> project(self_build LANGUAGES NONE)
```

**Comment by simi** [comment]
aka this change is missing in 2.6.2 - https://github.com/rubygems/rubygems/commit/5d05164c79f5f93e41100e33f5de5f96291b1dd2

**Comment by Heidistein** [comment]
I will test again with that change, however other things are going wrong now. A little time is required.

**Comment by Heidistein** [comment]
I confirm that by removing the word 'LANGUAGES' it indeed passes the tests.

**Comment by duckinator** [comment]
@simi thanks for finding the difference between the two environments! :+1:

**Comment by bronzdoc** [comment]
@Heidistein can this be closed?

**Comment by Heidistein** [comment]
Hmm. I am not sure. I leave it at your discretion.
Technically the latest stable is still broken, but the master is not.

I now know how to fix my build, for me it is not anymore so relevant.

**Comment by simi** [comment]
It is fixed in Ruby trunk and in rubygems master. I think this can be closed since it will be fixed in next Ruby release. Rubygems test suite shipped with Ruby 2.6.2 is not CMake 2 compatible by accident :'(

I'm not sure about supported CMake, but if we need to support CMake 2 as well, I can try to prepare some CI changes to get CMake 2 tested as well to prevent this situation in the future.
